src/position/template/position-presenter.template.ts

```diff
@@ -1,8 +1,37 @@
+import { Inject } from '@nestjs/common';
+
+import { APP_TOOLKIT, IAppToolkit } from '~app-toolkit/app-toolkit.interface';
+import { PresentationConfig } from '~app/app.interface';
+import { MetadataItemWithLabel } from '~balance/balance-fetcher.interface';
+import { ContractPositionBalance, TokenBalance } from '~position/position-balance.interface';
 import { Network } from '~types';
 
-export abstract class PositionPresenterTemplate {
+export interface PositionPresenter {
+  positionGroups?: PositionGroup[];
+  getBalanceProductGroups(): PositionGroup[] | null;
+  getExplorePresentation(): PresentationConfig | null;
+}
+
+export abstract class PositionPresenterTemplate implements PositionPresenter {
+  constructor(
+    @Inject(APP_TOOLKIT)
+    protected readonly appToolkit: IAppToolkit,
+  ) {}
+
   abstract network: Network;
   abstract appId: string;
+  positionGroups?: PositionGroup[];
+
+  getBalanceProductGroups(): PositionGroup[] | null {
+    return this.positionGroups ?? null;
+  }
 
-  positionGroups?: Record<string, string[]>;
+  getExplorePresentation(): PresentationConfig | null {
```

Comment:
Last is just make `explorePresentationConfig` as a member variable, and we're done.